backend/tracim_backend/lib/core/userworkspace.py

```diff
@@ -147,22 +156,42 @@ def create_one(
         return role
 
     def delete_one(self, user_id: int, workspace_id: int, flush=True) -> None:
-        if self._user and self._user.user_id == user_id:
-            raise UserCantRemoveHisOwnRoleInWorkspace(
-                "user {} can't remove is own role in workspace".format(user_id)
+        if self._is_last_workspace_manager(user_id, workspace_id):
+            raise LastWorkspaceManagerRoleCantBeModified(
+                "last workspace manager {} can't remove is own role in workspace".format(user_id)
             )
         role = self.get_one(user_id, workspace_id)
         self._session.delete(role)
         if flush:
             self._session.flush()
 
-    def get_all_for_workspace(self, workspace: Workspace) -> typing.List[UserRoleInWorkspace]:
+    def _is_last_workspace_manager(self, user_id: int, workspace_id: int) -> bool:
+        # INFO - G.M - 2020-17-09 - check if user is workspace_manager of workspace
+        try:
+            self.get_all_for_workspace_query(workspace_id).filter(
+                UserRoleInWorkspace.role == WorkspaceRoles.WORKSPACE_MANAGER.level
+            ).filter(UserRoleInWorkspace.user_id == user_id).one()
+        except NoResultFound:
+            return False
+
+        # INFO - G.M - 2020-17-09 - check other potential workspace_manager of workspace
+        return not bool(
```

Comment:
Wouldn't `return ....count()>0` be more readable ?

Reply:
Right that's more obvious.

Reply:
in fact, it should be `count() == 0`

Reply:
fixed